test/Base_Tests/src/Data/Time/Date_Spec.enso

```diff
@@ -552,7 +552,7 @@ js_date year month=1 day=1 =
 
 js_array_date year month=1 day=1 =
     arr = Panic.catch Any (js_array_dateCreate year month day) (err -> Error.throw (Time_Error.Error err.payload))
-    js_set_zone arr.at(0)
+    js_set_zone (arr.at 0)
```

Comment:
I do understand the change of syntax in this case, but the `Regex.compile('([a-c])')` seems different to me, because it is not about grouping the whole expression together. Here `js_set_zone arr.at 0` would mean something different than the expression with parenthesis. In the _regex case_ the expression with or without parenthesis still means the same.

Reply:
I think unspaced application needs to have unspaced precedence, because it *looks like* it binds more tightly than spaced application. So, I don't see any way to deny this while allowing the other cases without adding additional language complexity, for no user benefit that I can see.

Reply:
Are we now talking esthetics? Hard to argue with taste. I would probably not complained, if there wasn't a **prior art** showing that `Regex.compile('([a-c])')` was a **valid expression**. E.g. I am not talking about esthetics, but about **compatibility**.

My view is: It has always been the case that `(` and `)` _group things more tightly_ than any precedence. `(3 + 2)*5` is correct and yields the same result as `(3 + 2) * 5`

Letting `f x y` to mean the same as `f (x) (y)` and `f(x)(y)` causes _no harm_ and would be _compatible_ with current behavior.

Reply:
> I would probably not complain, if there wasn't a **prior art** showing that `Regex.compile('([a-c])')` was a **valid expression**. E.g. I am not talking about esthetics, but about **compatibility**.

Compatibility with what? This syntax has been used a few times in our repo, probably by accident. How much `.enso` code exists outside our repo? I think we have not yet run out of time to make some small compatibility-breaking changes.

> My view is: It has always been the case that `(` and `)` _group things more tightly_ than any precedence. `(3 + 2)*5` is correct and yields the same result as `(3 + 2) * 5`

That is not exactly what parens do, and this may be the heart of the issue. Parentheses group *their contents* more tightly than any precedence. The relationship of the parenthesized group with any surrounding expression depends on *the spacing outside the parentheses*.

If a series of terms don't have spaces between them (ignoring spacing inside the parentheses of any one of the terms), they form an unspaced group--which is reduced to a single term before being related to any surrounding spaced terms. We have no precedent for inlining an unspaced series of terms into a surrounding spaced expression. (Comparisons of value operators to value operators may cross unspaced groups, but: 1. If this causes inconsistency with unspaced-group precedence, a warning is produced; 2. Application is a functional operator.)
 
> Letting `f x y` to mean the same as `f (x) (y)` and `f(x)(y)` causes _no harm_ and would be _compatible_ with current behavior.

I think it's confusing. One of the few current usages of this syntax relied on the current interpretation, not the proposed interpretation.

> Argument with _unspaced precedence_ isn't valid as `(`+`)` have always taken a priority over spaces.

Parentheses take priority over spaces *inside* the group; if effect, parentheses *empower* the spacing *outside* the group, allowing such things as including a spaced expression on the inside to fit into an unspaced expression on the outside.

Reply:
> we have not yet run out of time to make some small compatibility-breaking changes

That's probably correct, but the time is running out. The sooner we get to compatibility mode, the less disruption we create among our customers in the months to come.

> parentheses empower the spacing outside the group, allowing such things as including a spaced expression on the inside to fit into an unspaced expression on the outside.

OK.